### DISCUSSION TITLE
non-availability of either meta or object db

### DISCUSSION THREAD

**Discussion by holygrolli** [juicedata/juicefs#2384] [Q&A] [answered]
Hi,

I didn’t find any explicit statement in any docs or discussion so here I am:

what happens if either the metadata db or the object db is unavailable for some time (simple network issues, updates etc). Will objects be cached in the cache dir or will the application experience file system errors and needs to deal with that?

I like the idea having unlimited cloud storage  for my private kubernetes project. While I hesitate setting up a quite expensive redundant cache infrastructure I also believe that just using a cloud provider’s sql db would be enough. If this single db is unavailable then my applications won’t work as well, will they?

Thanks for your help

**Comment by xiaogaozi** [comment, marked as answer]
When the metadata engine or object storage cannot be accessed normally, the JuiceFS client will automatically retry the request (refer to the `--io-retries` option). If the request fails after retries several times, an error ([`syscall.EIO`](https://pkg.go.dev/syscall#EIO)) will be returned to the application. Therefore, when your application encounters such an error, it needs to be handled accordingly.
